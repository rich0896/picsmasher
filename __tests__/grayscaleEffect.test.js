// __tests__/grayscaleEffect.test.js

const { GrayscaleEffect } = require('../effects.js'); // Adjust the path as needed
const { applyEffectAndGetImageData } = require('../helpers/testHelpers.js');

describe('GrayscaleEffect', () => {
    let ctx, canvas;

    beforeEach(() => {
        canvas = document.createElement('canvas');
        canvas.width = 10;
        canvas.height = 10;
        ctx = canvas.getContext('2d');

        // Fill the canvas with a solid color (e.g., rgb(100, 150, 200))
        ctx.fillStyle = 'rgb(100, 150, 200)';
        ctx.fillRect(0, 0, canvas.width, canvas.height);
    });

    test('should apply grayscale correctly at 100% intensity', () => {
        const effect = new GrayscaleEffect({ intensity: 100 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should handle zero intensity without changes', () => {
        const effect = new GrayscaleEffect({ intensity: 0 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });
});
