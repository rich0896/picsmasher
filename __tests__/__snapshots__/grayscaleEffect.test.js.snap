// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrayscaleEffect should apply grayscale correctly at 100% intensity 1`] = `ImageData(10, 10, [150, 150, 150, 255, 150, 150, 150, 255, 150, 150...])`;

exports[`GrayscaleEffect should handle zero intensity without changes 1`] = `ImageData(10, 10, [150, 150, 150, 255, 150, 150, 150, 255, 150, 150...])`;
